worklog.py: add -F/--follow option that follows the log

silkeh opened the pull request:
**Summary**

Add an option that prints build log entries as soon as they are available. This works similarly to the familiar options in `tail`.

**Test Plan**

```sh
./worklog.py builds -F now
```

**Checklist**

- (N/A) ~~Package was built and tested against unstable~~

ermo reviewed (approved):
I sort of like this. =)

silkeh commented:
@ermo: rebased! I also updated the script so the printing goes line by line, which is very useful for the `updates` command as you can now see the actual progress!

ermo reviewed (approved):
I see no reason this shouldn't go in as-is.

I do feel we need a note on the accepted date format with an example, but I am unsure how to do this with the current argparser setup...
